check_sd_repos.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+"""
+Check forked repoos for mergability with their upstreams. Designed to run as a
+scheduled task in buildbot.
+
+If there are merge conflicts, the problematic repos are printed and the script
+exits non-zero.
+"""
+
+from dataclasses import dataclass
+import subprocess
+import sys
+
+
+@dataclass
+class Repo:
+    name: str
+    url: str
+    branch: str
+    upstream_url: str
+    upstream_branch: str
+
+
+# Repositories to check.
+REPOS = [
+    Repo("ykrustc",
```

Comment:
Having a `Repo` struct but then unnamed parameters seems worse than just using a nested list to me (at least in terms of construction). At a minimum it might help to put each argument on a new line?

Reply:
The args are all named in the class. I just provided them in order.

Would you prefer if i prefixed each arg, e.g. `name=..., url=...`?

Reply:
I think named args would be clearer here: as soon as you get past about 3 args, I find remembering the order near-impossible.

Reply:
Fixed, but I need to wait until there are no CI jobs to test it properly.

Reply:
Unless you are confident that I made no mistakes :P 

FWIW: vim-lsp is happy with it.